src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -82,7 +81,28 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
                 if (!iDisposableConversion.IsImplicit)
                 {
                     TypeSymbol expressionType = expressionOpt.Type;
-                    if ((object)expressionType == null || !expressionType.IsErrorType())
+                    MethodSymbol disposeMethod = expressionType == null ? null : SatisfiesDisposePattern(expressionOpt.Type, diagnostics);
```

Comment:
Seems like this code should be outside the `iDisposable.IsImplicit` check. That is, we should do the pattern check *before* the conversion is classified, in case we don't need to do a conversion at all.